jupyterhub/files/hub/jupyterhub_config.py

```diff
@@ -323,24 +329,39 @@ def camelCaseify(s):
         "secretName": get_name("singleuser"),
         "items": items,
     }
-    c.KubeSpawner.volumes.append(volume)
+    volumes[volume["name"]] = volume
 
-    volume_mounts = []
-    for file_key, file_details in extra_files.items():
-        volume_mounts.append(
-            {
-                "mountPath": file_details["mountPath"],
-                "subPath": file_key,
-                "name": "files",
-            }
-        )
-    c.KubeSpawner.volume_mounts.extend(volume_mounts)
+    for idx, (file_key, file_details) in enumerate(extra_files.items()):
+        volume_mount = {
+            "mountPath": file_details["mountPath"],
+            "subPath": file_key,
+            "name": "files",
+        }
+        volume_mounts[f"{idx}-files"] = volume_mount
 
 # Inject extraVolumes / extraVolumeMounts
-c.KubeSpawner.volumes.extend(get_config("singleuser.storage.extraVolumes", []))
-c.KubeSpawner.volume_mounts.extend(
-    get_config("singleuser.storage.extraVolumeMounts", [])
-)
+extra_volumes = get_config("singleuser.storage.extraVolumes", default={})
+if isinstance(extra_volumes, dict):
+    for key, volume in extra_volumes.items():
+        volumes.update[key] = volume
+elif isinstance(extra_volumes, list):
+    for volume in extra_volumes:
```

Comment:
Should we defend against clobbering `files` or `home` here? This `extraVolumes` comes from the user values.